### PR TITLE
Feature/rewrite plugin webpack

### PR DESCRIPTION
**Pull request by dingo-d**
Ok so this is quite a big rewrite. In the light of core ticket [#40894](https://core.trac.wordpress.org/ticket/40894) I wanted to rewrite the plugin using webpack and more modern ES6 syntax of JS. Plus I wanted to separate business from presentational logic.

It is quite a bit away from your old plain jQuery, but I think that it's understandable enough and that it can encourage others in the WordPress community to start working with modern JS tools :)

I've made several modifications:

* Updated the readme with the build process
* Updated `package.json` with additional information, and dependencies
* Added ES linter
* Added babel for backward compatibility (needs testing on IE11)
* Cleaned up the code a bit, added proper messages that should be in the ajax response if something is wrong
* Changed `POST` to `GET`requests in the callbacks, since we are not writing anything to the database - it's semantically more correct

## To do:
* Replace css with sass in another PR

### CODE REVIEW

File: inc/routes.php [hunk 1]
```diff
@@ -13,58 +13,75 @@
...
+		$error = new WP_Error( '-1', $message );
+		wp_send_json_error( $error );
+	}
 
-	if ( isset( $_POST['hide_warning'] ) && 'true' === $_POST['hide_warning'] ) {
-		$args['show_warnings'] = true;
+	$theme_slug = sanitize_text_field( wp_unslash( $_GET['themeName'] ) );
+
+	if ( isset( $_GET['hideWarning'] ) && 'true' === $_GET['hideWarning'] ) {
+		$args['show_warnings'] = 'false'; // MANUAL INSPECTION!!!!!!!
```
Comment (grappler) [inline inc/routes.php]: What is the reason for the code comment?

File: inc/routes.php [hunk 2]
```diff
@@ -90,21 +107,29 @@ function theme_sniffer_run_sniffer() {
...
  *
+ * @param \WP_REST_Request $request Full data about the request.
  * @since 0.1.0
  */
-function theme_sniffer_individual_sniff() {
+function theme_sniffer_individual_sniff( \WP_REST_Request $request ) {
+	$headers = $request->get_headers();
+error_log( print_r( $_GET['themeName'], true ) );
+error_log( print_r( $_GET['themeArgs'], true ) );
+error_log( print_r( $_GET['file'], true ) );
```
Comment (grappler) [inline inc/routes.php]: Can you remove the debug code?

File: inc/routes.php (hunk 1 above)
Comment (dingo-d) [reply inc/routes.php]: Oh, my bad, forgot to remove this comment. I needed to check these `$_GET` variables. Will remove this.

File: inc/routes.php (hunk 2 above)
Comment (dingo-d) [reply inc/routes.php]: Will do.

**Comment by dingo-d** [comment]
@grappler @ernilambar did you have the time to look at the PR? I would like to start with namespacing the plugin, and maybe adding some unit tests if necessary.

**Comment by grappler** [comment]
@dingo-d Sorry for the delay. Thank you for working on this.

A couple a of comments and requests
-  Could we add the `ajaxStopped` text to the progress bar?
- When choosing a new theme while a check is running hides the progress bar.

**Comment by dingo-d** [comment]
@grappler Hi I made the changes you requested, should be better now. I made it so that when you switch the theme the checking will stop. Seems like ok behavior to me.

**Comment by dingo-d** [comment]
@grappler I fixed all the collisions, I should I try to clean the commit history with interactive rebase?

**Comment by dingo-d** [comment]
Thanks! I'll get on with other work asap!

**Pull request merged**